src/chart/generateCategoricalChart.tsx

```diff
@@ -2173,26 +2173,26 @@ export const generateCategoricalChart = ({
     }
 
     renderMap = {
-      CartesianGrid: { handler: renderAsIs, once: true },
-      ReferenceArea: { handler: this.renderReferenceElement },
-      ReferenceLine: { handler: renderAsIs },
-      ReferenceDot: { handler: this.renderReferenceElement },
-      XAxis: { handler: renderAsIs },
-      YAxis: { handler: renderAsIs },
-      Brush: { handler: this.renderBrush, once: true },
-      Bar: { handler: this.renderGraphicChild },
-      Line: { handler: this.renderGraphicChild },
-      Area: { handler: this.renderGraphicChild },
-      Radar: { handler: this.renderGraphicChild },
-      RadialBar: { handler: this.renderGraphicChild },
-      Scatter: { handler: this.renderGraphicChild },
-      Pie: { handler: this.renderGraphicChild },
-      Funnel: { handler: this.renderGraphicChild },
-      Tooltip: { handler: this.renderCursor, once: true },
-      PolarGrid: { handler: this.renderPolarGrid, once: true },
-      PolarAngleAxis: { handler: this.renderPolarAxis },
-      PolarRadiusAxis: { handler: this.renderPolarAxis },
-      Customized: { handler: this.renderCustomized },
+      CartesianGrid: { handler: renderAsIs, once: true, priority: 0 },
+      ReferenceArea: { handler: this.renderReferenceElement, priority: 1 },
+      ReferenceLine: { handler: renderAsIs, priority: 1 },
+      ReferenceDot: { handler: this.renderReferenceElement, priority: 1 },
+      XAxis: { handler: renderAsIs, priority: 0 },
+      YAxis: { handler: renderAsIs, priority: 0 },
+      Brush: { handler: this.renderBrush, once: true, priority: 1 },
+      Bar: { handler: this.renderGraphicChild, priority: 1 },
+      Line: { handler: this.renderGraphicChild, priority: 1 },
+      Area: { handler: this.renderGraphicChild, priority: 1 },
+      Radar: { handler: this.renderGraphicChild, priority: 1 },
+      RadialBar: { handler: this.renderGraphicChild, priority: 1 },
+      Scatter: { handler: this.renderGraphicChild, priority: 1 },
+      Pie: { handler: this.renderGraphicChild, priority: 1 },
+      Funnel: { handler: this.renderGraphicChild, priority: 1 },
+      Tooltip: { handler: this.renderCursor, once: true, priority: 0 },
+      PolarGrid: { handler: this.renderPolarGrid, once: true, priority: 0 },
+      PolarAngleAxis: { handler: this.renderPolarAxis, priority: 0 },
+      PolarRadiusAxis: { handler: this.renderPolarAxis, priority: 0 },
+      Customized: { handler: this.renderCustomized, priority: 1 },
```

Comment:
Why would we want to extend this with a number field, if really it can only be 0 or 1? 
It seems to me that what instead care for is: Foreground vs Background. 

I would always prefer types to be a simple as possible, and here `number` is needlessly open to extension. Soon, we might have someone add a new value, i.e. priority 2, or -1. 

What do you think? Could we implement this with only a boolean flag? Something like `isBackground: boolean`?